### PR TITLE
NickAkhmetov/CAT-74 Add pull request templates to ease future PR reviews

### PR DESCRIPTION
**Pull request by NickAkhmetov**
## Summary

This PR introduces pull request templates to add consistent structure to our pull requests and ensure there is enough detail for any of us to be able to review the changes at a glance.

## Issue Link

https://hms-dbmi.atlassian.net/browse/CAT-74

## Changes

There are four templates:
- Feature
	- New functionalities and iterations on previous functionalities; should correspond to a Story ticket in JIRA.
	- Package updates which impact a lot of existing code should also use this template.
- Bugfix
	- Fixes to broken functionalities; should correspond to a Bugfix ticket in JIRA.
- Documentation
	- Updates to documentation which do not touch site functionality
- Chore/other
	- Minor package upgrades which do not require changes to code, revisions to dev environment, ad hoc fixes/hotfixes.

## Impact

Using these templates should make it easier for us to refer back to previous work done and simplify the code review process; with that said, if this is not something our team desires to use, we can drop this idea.

## Additional notes
Templates from https://graphite.dev/guides/comprehensive-checklist-github-pr-template

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE/documentation.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+## Details
+
+Provide detailed information about the changes made to the documentation.
+
+## Motivation
+
+Explain why these documentation changes are necessary.
+
+## Review Checklist
```
Comment (john-conroy) [inline .github/PULL_REQUEST_TEMPLATE/documentation.md:17]: This section is `Review Checklist` where it's `Checklist` in the other templates, is that an intentional difference?

File: .github/PULL_REQUEST_TEMPLATE/documentation.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+List the sections or pages of the documentation that are updated.
+
+## Details
+
+Provide detailed information about the changes made to the documentation.
+
+## Motivation
+
+Explain why these documentation changes are necessary.
```
Comment (john-conroy) [inline .github/PULL_REQUEST_TEMPLATE/documentation.md:15]: Seems a bit superfluous to have 4 entries? I'm ok relying on code changes for some of the details. I think a single summary should suffice?

File: .github/PULL_REQUEST_TEMPLATE/documentation.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+## Review Checklist
+
+- [ ] Spelling and grammar are correct
```
Comment (john-conroy) [inline .github/PULL_REQUEST_TEMPLATE/documentation.md:19]: Should we suggest a means to evaluate grammar?

**Review by john-conroy — COMMENTED** [review]
I like the templates, but don't want to add a ton of overhead/boilerplate to the PRs. I feel like a lot of this information should be captured in the ticket already. Is it worthwhile to copy that information? If not, could we pare this down a bit?

**Comment by NickAkhmetov** [comment]
Making the following revisions:
- condensing to one simple template
- adding spellcheck extension recommendation

File: .github/PULL_REQUEST_TEMPLATE/other.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+Outline the changes made as part of this chore.
+
+## Impact
+
+Describe any implications this fix may have on other parts of the application.
+
+## Additional Notes
+
+Any further information needed to understand the fix or its impact.
```
Comment (lchoy) [inline .github/PULL_REQUEST_TEMPLATE/other.md:19]: These are good points. Similar to John's comment above, consider covering these in the Summary.

File: .github/PULL_REQUEST_TEMPLATE/feature.md [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+Describe how the feature has been tested, including both automated and manual testing strategies.
+
+## Screenshots/Video
+
+Include screenshots or video demonstrating the new feature, if applicable.
+
+## Checklist
+
+- [ ] Code follows the project's coding standards
```
Comment (lchoy) [inline .github/PULL_REQUEST_TEMPLATE/feature.md:31]: Are these formalized somewhere? There are some aspects in the README - Changelog files, linting

File: .github/PULL_REQUEST_TEMPLATE/feature.md [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+- [ ] Unit tests covering the new feature have been added
+
+- [ ] All existing tests pass
+
+- [ ] The documentation has been updated to reflect the new feature
```
Comment (lchoy) [inline .github/PULL_REQUEST_TEMPLATE/feature.md:37]: End user documentation doesn't exist, except lightly via info icons. Should we consider adding a section to the updated [documentation site](https://docs.hubmapconsortium.org/)?

**Review by lchoy — APPROVED** [review]

**Pull request merged**